packages/neo-one-server-client/src/ServerManager.ts

```diff
@@ -70,6 +49,31 @@ export class ServerManager {
     this.serverVersion = serverVersion;
   }
 
+  public async getRelativeServerVersion({
+    port,
+    expectedVersion,
+  }: {
+    readonly port: number;
+    readonly expectedVersion: string;
+  }): Promise<number | undefined> {
+    const client = new Client({ port });
+    const startTime = utils.nowSeconds();
+    // tslint:disable-next-line no-loop-statement
+    while (utils.nowSeconds() - startTime <= 5) {
+      try {
+        const version = await client.getVersion();
+
+        return version === expectedVersion ? 0 : semver.lt(version, expectedVersion) ? -1 : 1;
+      } catch {
+        const coldVersion = await this.getServerVersion();
```

Comment:
We should check for the server version in the file only after we've given up checking the server, i.e. outside of the `while` loop. No point in checking for a file that will never exist multiple times.

Reply:
👍